integ-test/src/test/java/org/opensearch/sql/ppl/TrendlineCommandIT.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.ppl;
+
+import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_BANK;
+import static org.opensearch.sql.util.MatcherUtils.rows;
+import static org.opensearch.sql.util.MatcherUtils.verifyDataRows;
+
+import java.io.IOException;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Test;
+
+public class TrendlineCommandIT extends PPLIntegTestCase {
+
+  @Override
+  public void init() throws IOException {
+    loadIndex(Index.BANK);
+  }
+
+  @Test
+  public void testTrendline() throws IOException {
+    final JSONObject result =
+        executeQuery(
+            String.format(
+                "source=%s | where balance > 39000 | sort balance | trendline sma(2, balance) as"
+                    + " balance_trend | fields balance_trend",
+                TEST_INDEX_BANK));
+    verifyDataRows(result, rows(new Object[] {null}), rows(44313.0), rows(39882.5));
+  }
+
+  @Test
+  public void testTrendlineMultipleFields() throws IOException {
+    final JSONObject result =
+        executeQuery(
+            String.format(
+                "source=%s | where balance > 39000 | sort balance | trendline sma(2, balance) as"
+                    + " balance_trend sma(2, account_number) as account_number_trend | fields"
+                    + " balance_trend, account_number_trend",
+                TEST_INDEX_BANK));
+    verifyDataRows(result, rows(null, null), rows(44313.0, 28.5), rows(39882.5, 13.0));
+  }
+
+  @Test
+  public void testTrendlineOverwritesExistingField() throws IOException {
+    final JSONObject result =
+        executeQuery(
+            String.format(
+                "source=%s | where balance > 39000 | sort balance | trendline sma(2, balance) as"
+                    + " age | fields age",
+                TEST_INDEX_BANK));
+    verifyDataRows(result, rows(new Object[] {null}), rows(44313.0), rows(39882.5));
+  }
```

Comment:
Please test error cases - `wma` and wrong data types.
How does it work if a column contains nulls or only nulls? A missing field/column case?

Reply:
If a column has a null it doesn't count as a sample (similar to how in SQL if you average a column and there are nulls they don't increment the divisor).

Reply:
I removed the WMA entry in the enum for now. Wrong data types are covered in the TrendlineOperatorTest unit test -- is it common to duplicate that type of test in integration?

Reply:
There are 3 reasons:
1. A lot of things mocked in UTs, so without IT you can't see an accumulated error (swiss cheese scenario).
2. IT can (and should) cover edge cases too.
3. You can easily convert IT to doctests. Once you have a dataset and queries, it is easy to add user docs/guides or record a demo using them.

Reply:
I'm going to hold off on adding ITs for edge/error cases as I think the unit tests catch these scenarios well enough.